### PR TITLE
[12] Fix s3 download and touch

### PR DESCRIPTION
**Pull request by icewind1991**
Backport of https://github.com/nextcloud/server/pull/7113

### CODE REVIEW

**Review by schiessle — APPROVED** [review]
I still get the same warning in preview generation but otherwise it works... Will open a separate issue for it.

File: lib/private/Files/ObjectStore/ObjectStoreStorage.php [hunk 1]
```diff
@@ -364,7 +364,7 @@ public function touch($path, $mtime = null) {
 			$fileId = $this->getCache()->put($path, $stat);
 			try {
 				//read an empty file from memory
```
Comment (nickvergessen) [inline lib/private/Files/ObjectStore/ObjectStoreStorage.php]: invalid

Comment (schiessle) [reply lib/private/Files/ObjectStore/ObjectStoreStorage.php]: right, can we agree to merge this one and then I fix the comment here: https://github.com/nextcloud/server/pull/7206 ? Then we have it fixed in master and stable12 in one run.

Comment (nickvergessen) [reply lib/private/Files/ObjectStore/ObjectStoreStorage.php]: fine by me

**Review by rullzer — APPROVED** [review]
:elephant:

**Pull request merged**